### PR TITLE
Trading

### PR DESCRIPTION
**Pull request by infobiac**

### CODE REVIEW

File: trading/models.py [hunk 1]
```diff
@@ -55,3 +55,13 @@ def __str__(self):
                                            self.quantity,
                                            self.account_id,
                                            self.stock_id)
+
+
+class TradeBucket(models.Model):
```
Comment (Neitsch) [inline trading/models.py]: TradeBucket and TradeStock have a lot in common

File: stocks/models.py [hunk 2]
```diff
@@ -202,6 +204,18 @@ def change_config(self, new_config):
...
+    def get_quote(self, date=None):
+        if date is None:
+            date = datetime.datetime.now()
+        stock_configs = self.get_stock_configs()
+        total = 0
+        for stock_config in stock_configs:
+            stock = stock_config.stock
+            price = stock.latest_quote(date).value
+            value = price * stock_config.quantity
+            total += value
```
Comment (Neitsch) [inline stocks/models.py:229]: Use the method on stock_config to calculate it instead of doing the math here

File: trading/models.py [hunk 3]
```diff
@@ -12,7 +12,7 @@ class TradingAccount(models.Model):
     A TradingAccount is owned by a user, we associate stock trades with it.
     """
     account_name = models.CharField(max_length=30)
-    profile = models.ForeignKey(Profile, related_name='trading_accounts')
+    profile = models.ForeignKey(Profile, related_name='trading_account')
```
Comment (Neitsch) [inline trading/models.py:14]: ?

File: trading/views.py [hunk 4]
```diff
@@ -2,15 +2,160 @@
...
+from django.core.exceptions import ObjectDoesNotExist
 from stocks.models import InvestmentBucket
+from trading.models import TradingAccount, TradeBucket
+from authentication.views import get_balance_as_dic
+from collections import defaultdict
 
 
-def test(request):
+@login_required
+def create_account(request):
```
Comment (Neitsch) [inline trading/views.py:15]: Let's just make a default trading account when the profile is created

File: trading/views.py [hunk 5]
```diff
@@ -2,15 +2,160 @@
...
+    return dict((k, v) for k, v in avail_buckets.items() if v > 0)
+
+
+@login_required
+def buy_bucket(request):
+    """
+    Display buy bucket, allow purchasing of bucket
+    """
+    if request.method == "GET":
+        buckets = get_bucket_stock_prices(InvestmentBucket.objects.all())
```
Comment (Neitsch) [inline trading/views.py:105]: InvestmentBucket.objects.filter(Q(owner=request.user.profile) | Q(public=True)).all()

File: trading/views.py [hunk 6]
```diff
@@ -2,15 +2,160 @@
...
+        context = {'buckets': buckets, 'verb': 'Buy'}
+        return render(request, template, context)
+    elif request.method == "POST":
+        prof = request.user.profile
+        try:
+            account = TradingAccount.objects.get(profile=prof)
+            bucketid = request.POST.get("bucket")
+            if bucketid is None:
+                return HttpResponse("You must specify a bucket")
+            bucket = InvestmentBucket.objects.get(pk=bucketid)
```
Comment (Neitsch) [inline trading/views.py:116]: InvestmentBucket.objects.filter(Q(owner=request.user.profile) | Q(public=True)).get(pk=bucketid)

File: trading/views.py [hunk 7]
```diff
@@ -2,15 +2,160 @@
...
+            if get_available_cash_as_num(request) > bucket.get_quote():
+                new_trade = TradeBucket(account=account, quantity=1, bucket=bucket)
```
Comment (Neitsch) [inline trading/views.py:118]: why 1?

File: trading/views.py [hunk 8]
```diff
@@ -2,15 +2,160 @@
...
+        context = {'buckets': buckets, 'verb': 'Sell'}
+        return render(request, template, context)
+    elif request.method == "POST":
+        prof = request.user.profile
+        account = TradingAccount.objects.get(profile=prof)
+        bucketid = request.POST.get("bucket")
+        if bucketid is None:
+            return HttpResponse("You must specify a bucket")
+        bucket = InvestmentBucket.objects.get(pk=bucketid)
+        new_trade = TradeBucket(account=account, quantity=-1, bucket=bucket)
```
Comment (Neitsch) [inline trading/views.py:147]: Why 1?

**Review by Neitsch — CHANGES_REQUESTED** [review]

File: trading/models.py (hunk 3 above)
Comment (infobiac) [reply trading/models.py:14]: think this must be a merging issue...

File: trading/views.py (hunk 4 above)
Comment (infobiac) [reply trading/views.py:15]: and not allow further accounts at all, or in addition to?

File: trading/views.py (hunk 5 above)
Comment (infobiac) [reply trading/views.py:105]: makes sense, will change.

File: trading/views.py (hunk 6 above)
Comment (infobiac) [reply trading/views.py:116]: is the filter just to make sure user has access, or what?

File: trading/views.py (hunk 4 above)
Comment (Neitsch) [reply trading/views.py:15]: For now, no further creation. Once we have solidified the model, we can expand into multiple

File: trading/views.py (hunk 6 above)
Comment (Neitsch) [reply trading/views.py:116]: Correct

**Comment by Neitsch** [comment]
Does the PR hold any advantage over #103 ?

**Comment by Neitsch** [comment]
Can I close it?

**Pull request closed without merging**